### PR TITLE
Remove deprecated features in 1.7

### PR DESCRIPTION
**Pull request by tk0miya**
Remove deprecated features (for v1.7!)

### CODE REVIEW

**Comment by tk0miya** [comment]
@jfbu could you check https://github.com/sphinx-doc/sphinx/pull/3677/commits/361a34257becbc0ae87c6ae10609427dab7ab11f is enough (dropping `latex_keep_old_macro_names`)?

**Review by jfbu — APPROVED** [review]
LGTM regarding latex_keep_old_macro_names! (with nits, see comments nexts on my two oversights)

File: doc/latex.rst [hunk 1]
```diff
@@ -414,10 +414,7 @@ Let us now list some macros from the package file
 - text styling commands (they have one argument): ``\sphinx<foo>`` with
   ``<foo>`` being one of ``strong``, ``bfcode``, ``email``, ``tablecontinued``,
   ``titleref``, ``menuselection``, ``accelerator``, ``crossref``, ``termref``,
-  ``optional``. The non-prefixed macros will still be defined if option
-  :confval:`latex_keep_old_macro_names` has been set to ``True`` (default is
-  ``False``), in which case the prefixed macros expand to the
-  non-prefixed ones.
+  ``optional``.
 
   .. versionadded:: 1.4.5
```
Comment (jfbu) [inline doc/latex.rst]: oh oversight in my review these lines should be either removed or modified to simply state that in the past the macros were not prefixed with `\sphinx` but now they are (at 1.7)

**Review by jfbu — COMMENTED** [review]
again I forgot to mention comment line at https://github.com/sphinx-doc/sphinx/pull/3677/commits/361a34257becbc0ae87c6ae10609427dab7ab11f#diff-f6432e57ae706b42894b186ffe699da0R1313

should now be removed

File: doc/latex.rst (hunk 1 above)
Comment (jfbu) [reply doc/latex.rst]: I meant the lines starting at https://github.com/sphinx-doc/sphinx/pull/3677#discussion-diff-114015939R419 and next

Comment (tk0miya) [reply doc/latex.rst]: Thanks, I'd overlooked too. I'd removed.
Thank you for reviewing!

**Review by jfbu — APPROVED** [review]
LGTM!

**Review by shimizukawa — APPROVED** [review]
LGTM!

**Pull request merged**